src/components/studentForm/StudentForm.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Select from 'react-select';
+import { useForm, Controller } from 'react-hook-form';
+import Input from '@mui/material/Input';
+import Checkbox from '@mui/material/Checkbox';
+import { Input as AntdInput } from 'antd';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+
+import './style.css';
+
+const schema = yup
+  .object({
+    name: yup.string().required(),
+    marks: yup.number().positive().integer().required(),
+    sunject: yup.string().required(),
+    subject: yup.string().required(),
+    grade: yup.string().required(),
+  })
+  .required();
```

Comment:
Keep validation schema in a separate file. This file should contain component things.